media-sound/clementine: `one fix for releases and a couple enhancements for the live

Fat-Zer opened the pull request:
It reveals that release clementine versions require udiks:0 rather than udisks:2
The support for the later was added in live;
Also some more enhancements to the live ebuild according to changes in upsream...

gktrk commented:
@gentoo/proxy-maint

gktrk commented:
Hi, runtime dependency slot changes require revision bumps to avoid vdb breakage.

Fat-Zer commented:
May I move ebuilds to revbump rather than copy? and is it ok to add them with stable keywords?

gktrk commented:
Unfortunately not :( you are moving from slot 0 to slot 2 for a dep. no reason to risk breaking the stable. let's revbump without removal and with ~arch keywords.

Fat-Zer commented:
Ok, I've added a fix for 1.2.3 (the stable one) as a new revbump with ~arch uses and moved 1.3.1(~) to -r1.

> you are moving from slot 0 to slot 2 for a dep

Note that it's vice-versa... the slot 2 was actually wrong one here.

gktrk commented:
> Note that it's vice-versa... the slot 2 was actually wrong one here.

revbump is even more important then to propagate the change to the vdb

Fat-Zer commented:
> revbump is even more important then to propagate the change to the vdb

What is vdb?

gktrk commented:
> What is vdb?

It is the installed packages database. When a package manager calculates dependencies, it looks at the vdb to determine the dependencies for the installed packages. If you modify a runtime dependency for an ebuild, the package manager will not pick it up because it will use the one recorded in the vdb which lists the old deps.

If you revbump, the package manager will see it as an upgrade, install the revbumped version and record the correct dependencies in the vdb.

Hope this makes things a bit more clear.

Fat-Zer commented:
thanks, I was aware of /var/db/pkg, but met the term «vdb» for the first time...

SoapGentoo commented on media-sound/clementine/clementine-1.2.3-r1.ebuild:
```diff
@@ -0,0 +1,169 @@
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Please consider porting this revbump to `EAPI=6` as well.

Fat-Zer replied:
1.2.3 will be removed sooner or later. IMHO no matter to enhance it beyond the bug fixing...

SoapGentoo replied:
fair enough

SoapGentoo commented on media-sound/clementine/clementine-1.2.3-r1.ebuild:
```diff
@@ -0,0 +1,169 @@
+	media-plugins/gst-plugins-taglib:0.10
+"
+DEPEND="${COMMON_DEPEND}
+	>=dev-libs/boost-1.39
```
dev-libs/boost **always** requires sub-slot operator `:=`

SoapGentoo commented on media-sound/clementine/clementine-1.2.3-r1.ebuild:
```diff
@@ -0,0 +1,169 @@
+		-DBUILD_WERROR=OFF
+		-DLINGUAS="${langs}"
+		-DBUNDLE_PROJECTM_PRESETS=OFF
+		$(cmake-utils_use cdda ENABLE_AUDIOCD)
```
if you do bump this to `EAPI=6`, you'll have to replace all these `cmake-utils_use` functions. While this might be slightly monotonous, it will help us to eventually slim down `cmake-utils.eclass`!

Fat-Zer commented:
fixed the boost subslot issue
removed dep upon media-libs/libechonest due to it is not required anymore

gktrk commented:
Are we good to go here?

SoapGentoo commented:
Fine from my side

gktrk commented:
Merged in https://github.com/gentoo/gentoo/commit/5fed2e48db334156e3241ed5e0f6bef75ae30a1f. Thanks!

I mangled the commit message for one of the commits because it wasn't following the proper [formatting rules](https://devmanual.gentoo.org/ebuild-maintenance/index.html#git-commit-message-format). We don't include ${PV} as in `media-sound/clementine-9999`.
